modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java

```diff
@@ -1063,7 +1135,13 @@ private GridThreadLocalTable fakeTable(Connection c, int idx) {
             List<ClusterNode> owners = cctx.topology().owners(p);
 
             if (F.isEmpty(owners)) {
-                if (!F.isEmpty(dataNodes(cctx.name(), NONE)))
+                // Handle special case: no mapping is configured for a partition.
+                if (F.isEmpty(cctx.affinity().assignment(NONE).get(p))) {
+                    partLocs[p] = Collections.emptySet(); // Mark unmapped partition.
```

Comment:
Lets add a constant `UNMAPPED_PARTS` for `Collections.emptySet()` and use it everywhere.